scripts/pi-hole/js/gravity.js

```diff
@@ -47,9 +47,18 @@ function eventsource() {
               // Enqueue the next data chunk into our target stream
               controller.enqueue(value);
               var string = new TextDecoder().decode(value);
-              // Remove ${OVER} from the string
-              string = string.replaceAll("\r[K", "\n");
-              ta.append(string);
+
+              // If a Carriage Return is found ...
+              if (string.indexOf("\r") === -1) {
```

Comment:
Isn't this functionally equivalent to the easier to read/understand
```suggestion
              if (string[0] !== '\r') {
```
?

See also my main review comment why I think this is the case.

Reply:
They are not the same:

`string.indexOf("\r") === -1` will be `false` if `\r` is found in the string (in any position).

Your suggestion will be `false` only if the first character is `\r`.

```js
var string = "test\r";
if (string.indexOf("\r") === -1) {
  console.log("CR not found");
} else {
  console.log("CR found at "+string.indexOf("\r"));
}
```

Output: `CR found at 4`.

Reply:
Sure that, but your example even underlines my point :-)

Assume we already have
```
abc
def
ghi
```
and not receive your chunk `test\r`.

The trailing `\r` would trigger removing `ghi` even though this is completely wrong.

With my suggested change, however, the code would *not* remove `ghi` and the output would become:
```
abc
def
ghi
test
```

IMO this is the correct behavior.